axon-server-connector/src/main/java/org/axonframework/axonserver/connector/event/EventConverter.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.axonserver.connector.event;
+
+import com.google.protobuf.ByteString;
+import io.axoniq.axonserver.grpc.event.dcb.Event;
+import io.axoniq.axonserver.grpc.event.dcb.TaggedEvent;
+import jakarta.annotation.Nonnull;
+import org.axonframework.common.infra.ComponentDescriptor;
+import org.axonframework.common.infra.DescribableComponent;
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.eventhandling.GenericEventMessage;
+import org.axonframework.eventsourcing.eventstore.Tag;
+import org.axonframework.eventsourcing.eventstore.TaggedEventMessage;
+import org.axonframework.messaging.MessageType;
+import org.axonframework.messaging.MetaData;
+import org.axonframework.serialization.Converter;
+
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Wrapper around standard Axon Framework {@link Converter} that can convert
+ * {@link TaggedEventMessage TaggedEventMessages} to {@link TaggedEvent TaggedEvents}.
+ *
+ * @author Marc Gathier
+ * @author Steven van Beelen
+ * @since 4.0.0
```

Comment:
I am unable to follow the version logic sometimes. Why is this 4.0.0? I am looking in the Axon server connector that exists, but I don't see anything similar.  The closest thing I can find is the `GrpcMetaDataConverter`, but that is conceptually very different. 

I am starting to think we should draw a very clear line between what constitutes as new and what as update. Anyone looking at this will assume this existed in 4.0.0, but won't be able to find it. I am starting to think that if it's a new class, it should be the current version :thinking:

Reply:
Aah, sorry, this was a leftover when I took the `GrpcMetaDataConverter` as a basis. Dropped the since and author tag just now.